### PR TITLE
Make request to set item status

### PR DESCRIPTION
**Pull request by ktp242**
This PR adds the functions for the app to send a request to set an item's status. The user now can type in a barcode to the from, press submit, and update the status. It also has the validation. Based on Kate's request, the barcodes for refile could have different digits(not only 14), nor characters(not only numbers). As a result, we only check if the barcodes are less than 20 digits.

### CODE REVIEW

File: src/shared/containers/Refile/ClearItemStatusForm.jsx [hunk 1]
```diff
@@ -144,12 +129,20 @@ class ClearItemStatusForm extends Component {
   handleFormSubmit(event) {
     event.preventDefault();
 
+    // The array that stores the results after validating the date inputs
```
Comment (thisisstephenbetts) [inline src/shared/containers/Refile/ClearItemStatusForm.jsx]: It'll be all the inputs, right? Not _just_ the date inputs?

File: src/shared/utils/ValidationUtils.js [hunk 2]
```diff
@@ -5,3 +5,36 @@ export function isBarcodeValid(barcode, requiredLength = 14) {
...
+/**
+* @desc Validates the input
+* @param {string} date - the string of the input value
+*/
+export function isDateValid(dateInput) {
+  if (!dateInput || typeof dateInput !== 'string') {
+    return false;
+  }
+
+  const dateArray = dateInput.split('-');
```
Comment (thisisstephenbetts) [inline src/shared/utils/ValidationUtils.js]: This all works and everything, but it feels a bit like you're mixing up the regex approach, and a more direct `split`-based approach.

One way to DRY it out a little bit would be to remove line 18, and then move lines 19 and 20 until after checking the result of the regex match (currently lines 25-27) and replace 
`const month = parseInt(dateArray[1], 10);` with 
`const month = parseInt(dateMatches[2], 10);` 
and similarly
`const date = parseInt(dateArray[2], 10);` with
`const date = parseInt(dateMatches[3], 10);`

File: src/shared/utils/ValidationUtils.js [hunk 3]
```diff
@@ -5,3 +5,36 @@ export function isBarcodeValid(barcode, requiredLength = 14) {
...
+  const month = parseInt(dateArray[1], 10);
+  const date = parseInt(dateArray[2], 10);
+  // Checks if it has a valid date format. The Regex check if the inputs are digits
+  // and if they have right number of digits
+  const dateMatches = dateInput.match(/^(\d{4})\-(\d{2})\-(?:\d{2})$/);
```
Comment (thisisstephenbetts) [inline src/shared/utils/ValidationUtils.js]: If you make the changes you will need to change the regex from 
`/^(\d{4})\-(\d{2})\-(?:\d{2})$/` to 
`/^(\d{4})\-(\d{2})\-(\d{2})$/`

(i.e., turning a non-matching group to a matching group — a matching group is available in the `dateMatches` array, whereas a non-matching group is just for organization)

**Review by thisisstephenbetts — APPROVED** [review]
Tiny suggestion about cleaning up the date validation function, but looks good!

File: src/shared/containers/Refile/ClearItemStatusForm.jsx (hunk 1 above)
Comment (ktp242) [reply src/shared/containers/Refile/ClearItemStatusForm.jsx]: Yeah! Thanks for correcting!

File: src/shared/utils/ValidationUtils.js (hunk 2 above)
Comment (ktp242) [reply src/shared/utils/ValidationUtils.js]: It is a great approach! Thanks!

**Pull request merged**